client/tests/integration/events/pipeline.rs

```diff
@@ -82,12 +82,12 @@ impl Checker {
         thread::spawn(move || {
             let mut event_iterator = self
                 .listener
-                .listen_for_events(FilterBox::Pipeline(
+                .listen_for_events(
                     PipelineEventFilter::new()
-                        .entity_kind(PipelineEntityKind::Transaction)
-                        .status_kind(status_kind)
-                        .hash(*self.hash),
-                ))
+                        .from_entity_of_kind(PipelineEntityKind::Transaction)
+                        .with_status(status_kind)
+                        .from_entity_with_hash(*self.hash),
```

Comment:
```suggestion
                        .with_hash(*self.hash),
```

might be more appropriate

Reply:
or 
```suggestion
                        .for_hash(*self.hash),
```